src/bpm/runc/adapter/adapter_test.go

```diff
@@ -453,6 +453,36 @@ var _ = Describe("RuncAdapter", func() {
 			Expect(os.RemoveAll(resolvConfDir)).To(Succeed())
 		})
 
+		It("sorts the mounts by the number of elments in their path", func() {
+			procCfg.AdditionalVolumes = []config.Volume{
+				{Path: "/path/with/six/elements/1/2"},
+				{Path: "/path/with/seven/elements/1/2/3"},
+				{Path: "/path/with/eight/elements/1/2/3/4"},
+			}
+
+			spec, err := runcAdapter.BuildSpec(logger, bpmCfg, procCfg, user)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(spec.Mounts).To(HaveLen(23))
+			Expect(spec.Mounts[23-1]).To(Equal(specs.Mount{
+				Destination: "/path/with/eight/elements/1/2/3/4",
+				Type:        "bind",
+				Source:      "/path/with/eight/elements/1/2/3/4",
+				Options:     []string{"nodev", "nosuid", "noexec", "rbind", "ro"},
+			}))
+			Expect(spec.Mounts[23-2]).To(Equal(specs.Mount{
+				Destination: "/path/with/seven/elements/1/2/3",
+				Type:        "bind",
+				Source:      "/path/with/seven/elements/1/2/3",
+				Options:     []string{"nodev", "nosuid", "noexec", "rbind", "ro"},
+			}))
+			Expect(spec.Mounts[23-3]).To(Equal(specs.Mount{
+				Destination: "/path/with/six/elements/1/2",
+				Type:        "bind",
+				Source:      "/path/with/six/elements/1/2",
+				Options:     []string{"nodev", "nosuid", "noexec", "rbind", "ro"},
+			}))
```

Comment:
I was trying to avoid creating a typed slice for sorting by taking advantage of the `sort.Slice` function. I'll give it a once over and see if there is an easy way to use a matcher, because that would be easier.

Reply:
https://golang.org/pkg/sort/#SliceIsSorted Clearly I can't read.